airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/spec/DeclarativeSpecificationFactory.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.spec
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import io.airbyte.cdk.command.ValidatedJsonUtils
+import io.airbyte.cdk.load.command.dlq.ObjectStorageSpec
+import io.airbyte.cdk.load.model.spec.Spec
+import io.airbyte.cdk.spec.SpecificationFactory
+import io.airbyte.protocol.models.v0.ConnectorSpecification
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+
+class DeclarativeSpecificationFactory(private val declarativeSpec: Spec) : SpecificationFactory {
+    override fun create(): ConnectorSpecification {
+        val connectionSpecificationCopy: ObjectNode =
+            declarativeSpec.connectionSpecification.deepCopy()
+        appendObjectStorageConfig(connectionSpecificationCopy)
+
+        val spec =
+            ConnectorSpecification()
+                .withConnectionSpecification(connectionSpecificationCopy)
+                // FIXME do we really need the following. I added those for backward compabitiliy
```

Comment:
Does anyone know if these are needed? I think @gosusnp added these initially in destination-salesforce template that he created

Reply:
good question, this pre-dates me, I kept it because it was everywhere else and I was focusing on something else at that time.